prov/usnic/src/usdf_fabric.c

```diff
@@ -101,6 +101,13 @@ usdf_validate_hints(struct fi_info *hints, struct usd_device_attrs *dap)
 		return -FI_ENODATA;
 	}
 
+	/* it doesn't make sense to request both prefix and prefix2 */
+	if ((hints->mode & (FI_MSG_PREFIX | FI_MSG_PREFIX2)) ==
+			(FI_MSG_PREFIX | FI_MSG_PREFIX2)) {
+		USDF_INFO("Simultaneous use of FI_MSG_PREFIX and FI_MSG_PREFIX2 is invalid.\n");
+		return -EINVAL;
+	}
```

Comment:
For clarification, are you suggesting that `FI_MSG_PREFIX2` be hidden and the applications version determine whether `FI_MSG_PREFIX2` or `FI_MSG_PREFIX` is set when FI_MSG_PREFIX is requested?

One of the issues there is that we only have `fi_getinfo` to handle version information. So that has the caveat that the app would be required to use `fi_getinfo` and when it requests `FI_MSG_PREFIX` the output might have a different mode bit set. That might confuse the app...